src/components/top_bar/top_bar.ts

```diff
@@ -291,4 +289,228 @@ export class TopBar extends Component<Props, SpreadsheetChildEnv> {
+  .addChild("textStyle", {
+    component: ActionButton,
+    props: {
+      action: ACTION_FORMAT.formatBold,
+      class: "o-hoverable-button o-toolbar-button",
+    },
+    sequence: 1,
+  })
+  .addChild("textStyle", {
+    component: ActionButton,
+    props: {
+      action: ACTION_FORMAT.formatItalic,
+      class: "o-hoverable-button o-toolbar-button",
+    },
+    sequence: 2,
+  })
+  .addChild("textStyle", {
+    component: ActionButton,
+    props: {
+      action: ACTION_FORMAT.formatStrikethrough,
+      class: "o-hoverable-button o-toolbar-button",
+    },
+    sequence: 3,
+  })
+  .addChild("textStyle", {
+    component: TopBarTextColorEditor,
+    props: {
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+      style: "textColor",
+      icon: "o-spreadsheet-Icon.TEXT_COLOR",
+    },
+    sequence: 4,
+  })
+  .add("cellStyle")
+  .addChild("cellStyle", {
+    component: TopBarFillColorEditor,
+    props: {
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+      style: "fillColor",
+      icon: "o-spreadsheet-Icon.FILL_COLOR",
+    },
+    sequence: 1,
+  })
+  .addChild("cellStyle", {
+    component: BorderEditorWidget,
+    props: {
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+    },
+    sequence: 2,
+  })
+  .addChild("cellStyle", {
+    component: ActionButton,
+    props: {
+      action: ACTION_EDIT.mergeCells,
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+    },
+    sequence: 3,
+  })
+  .add("alignment")
+  .addChild("alignment", {
+    component: DropdownAction,
+    props: {
+      parentAction: ACTION_FORMAT.formatAlignmentHorizontal,
+      childActions: [
+        ACTION_FORMAT.formatAlignmentLeft,
+        ACTION_FORMAT.formatAlignmentCenter,
+        ACTION_FORMAT.formatAlignmentRight,
+      ],
+      class: "o-hoverable-button o-toolbar-button",
+      childClass: "o-hoverable-button",
+    },
+    sequence: 1,
+  })
+  .addChild("alignment", {
+    component: DropdownAction,
+    props: {
+      parentAction: ACTION_FORMAT.formatAlignmentVertical,
+      childActions: [
+        ACTION_FORMAT.formatAlignmentTop,
+        ACTION_FORMAT.formatAlignmentMiddle,
+        ACTION_FORMAT.formatAlignmentBottom,
+      ],
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+      childClass: "o-hoverable-button",
+    },
+    sequence: 2,
+  })
+  .addChild("alignment", {
+    component: DropdownAction,
+    props: {
+      parentAction: ACTION_FORMAT.formatWrapping,
+      childActions: [
+        ACTION_FORMAT.formatWrappingOverflow,
+        ACTION_FORMAT.formatWrappingWrap,
+        ACTION_FORMAT.formatWrappingClip,
+      ],
+      class: "o-hoverable-button o-menu-item-button o-toolbar-button",
+      childClass: "o-hoverable-button",
+    },
+    sequence: 3,
+  })
+  .add("misc")
+  .addChild("misc", {
+    component: TableDropdownButton,
+    props: { class: "toolbar-button" },
```

Comment:
`o-toolbar-button` ? 
I'm pretty sure that it's the only element that has not the `o-hoverable-button`, is it expected ?
If it's not intended, I would move these two classes (`o-hoverable-button` and `o-toolbar-button`) inside the template, and use a div or a span with these classes in the t-foreach ?

Reply:
it's been a battle already to use it because some of those components alreay contain some classes that make it quite hard to work. The hoverable button class was designed to interact with o-menu-item, and that class is present in some of the sub component (like ActionButton)
I've ben trying for an hour an can't make it work don't know if it's worth it